script/import_scripts/vbulletin.rb

```diff
@@ -140,20 +149,26 @@ def import_users
           id: user["userid"],
           name: username,
           username: username,
+          password: user["crypted_password"],
           email: email,
           website: user["homepage"].strip,
           title: @htmlentities.decode(user["usertitle"]).strip,
           primary_group_id: group_id_from_imported_group_id(user["usergroupid"].to_i),
           created_at: parse_timestamp(user["joindate"]),
           last_seen_at: parse_timestamp(user["lastvisit"]),
           post_create_action: proc do |u|
-            @old_username_to_new_usernames[user["username"]] = u.username
             import_profile_picture(user, u)
             import_profile_background(user, u)
           end
         }
       end
     end
+
+    @usernames = {}
+    UserCustomField.where(name: 'import_username').pluck(:user_id, :value).each do |user_id, import_username|
+      @usernames[import_username] = user_id
```

Comment:
I guess you moved the username mapping from `post_create_action` to this location in order to make the importer resumable or was there another reason? But why store the `user_id` instead of the username itself? The following should work...

```ruby
@usernames = UserCustomField.joins(:user).where(name: 'import_username').pluck('user_custom_fields.value', 'users.username').to_h
```

Reply:
Lazyness in two senses:
- as in: lazy programmer
- but also as in lazy evaluation: delay user lookup until it is actually needed (and yes one join may be faster than 100 separate lookups, and that is where reason #1 comes into play)

Reply:
Would you mind not being a lazy programmer in this case? 😜 
I'd appreciate it if you could update your PR so that it uses a join for loading the usernames.

Reply:
We already finished this import and I think the performance impact is negligible so I'm not really eager to revisit and test this all over again, especially since this is already working correctly.

Reply:
Seems we have to do some fixes anyway so yeah I will change this while I'm at it, coming up in a few days, please keep this open in the meanwhile

Reply:
Sure and thanks for fixing the importer! 👍